src/components/views/messages/DateSeparator.tsx

```diff
@@ -62,12 +106,119 @@ export default class DateSeparator extends React.Component<IProps> {
+            // detached queue and we show the room status bar to allow retry
+            if (typeof code !== "undefined") {
+                // display error message stating you couldn't delete this.
+                Modal.createTrackedDialog('Unable to find event at that date', '', ErrorDialog, {
+                    title: _t('Error'),
+                    description: _t('Unable to find event at that date. (%(code)s)', { code }),
+                });
+            }
+        }
+    };
+
+
+    private onLastWeekClicked = (): void => {
+        const date = new Date();
+        // This just goes back 7 days.
+        // FIXME: Do we want this to go back to the last Sunday? https://upokary.com/how-to-get-last-monday-or-last-friday-or-any-last-day-in-javascript/
+        date.setDate(date.getDate() - 7);
+        this.pickDate(date);
+        this.closeMenu();
+    }
+
+    private onLastMonthClicked = (): void => {
+        const date = new Date();
+        // Month numbers are 0 - 11 and `setMonth` handles the negative rollover
+        date.setMonth(date.getMonth() - 1, 1);
+        this.pickDate(date);
+        this.closeMenu();
+    }
+
+    private onTheBeginningClicked = (): void => {
+        const date = new Date(0);
+        this.pickDate(date);
+        this.closeMenu();
+    }
+
+    private onDatePicked = (dateString): void => {
+        this.pickDate(dateString);
+        this.closeMenu();
+    }
+
+    private renderNotificationsMenu(): React.ReactElement {
+        let contextMenu: JSX.Element;
+        if (this.state.contextMenuPosition) {
+            contextMenu = <IconizedContextMenu
+                {...contextMenuBelow(this.state.contextMenuPosition)}
+                compact
+                onFinished={this.onContextMenuCloseClick}
+            >
+                <IconizedContextMenuOptionList first>
+                    <IconizedContextMenuOption
+                        label={_t("Last week")}
+                        onClick={this.onLastWeekClicked}
+                    />
+                    <IconizedContextMenuOption
+                        label={_t("Last month")}
+                        onClick={this.onLastMonthClicked}
+                    />
+                    <IconizedContextMenuOption
+                        label={_t("The beginning of the room")}
+                        onClick={this.onTheBeginningClicked}
+                    />
+                </IconizedContextMenuOptionList>
+
+                <IconizedContextMenuOptionList>
+                    <JumpToDatePicker ts={this.props.ts} onDatePicked={this.onDatePicked} />
+                </IconizedContextMenuOptionList>
+            </IconizedContextMenu>;
+        }
+
+        return (
+            <ContextMenuTooltipButton
+                className="mx_DateSeparator_jumpToDateMenu"
+                onClick={this.onContextMenuOpenClick}
+                isExpanded={!!this.state.contextMenuPosition}
+                title={_t("Jump to date")}
```

Comment:
TODO: Why does the tooltip decide to overlap the element?

<img width="168" alt="" src="https://user-images.githubusercontent.com/558581/145678377-0e812293-e4c9-4167-9f02-8b7fa8176b8c.png">

<img width="159" alt="" src="https://user-images.githubusercontent.com/558581/145678379-8e0b77ac-2bf4-456f-828b-39f42a1e4dd7.png">

Reply:
This is being addressed in https://github.com/matrix-org/matrix-react-sdk/pull/7551